beacon/handlers.go

```diff
@@ -510,11 +502,45 @@ func (pd *ProtocolDriver) registerProposed(logger log.Log, epoch types.EpochID,
 	return pd.states[epoch].registerProposed(logger, nodeID)
 }
 
-func (pd *ProtocolDriver) registerVoted(logger log.Log, epoch types.EpochID, nodeID types.NodeID, round types.RoundID) error {
+func (pd *ProtocolDriver) registerVoted(epoch types.EpochID, nodeID types.NodeID, round types.RoundID) error {
 	pd.mu.Lock()
 	defer pd.mu.Unlock()
 	if _, ok := pd.states[epoch]; !ok {
 		return errEpochNotActive
 	}
-	return pd.states[epoch].registerVoted(logger, nodeID, round)
+	return pd.states[epoch].registerVoted(nodeID, round)
+}
+
+const wordSize = 64
+
+func newVotesTracker(limit uint32) *votesTracker {
+	b := bits.Len32(limit)
+	words := b / wordSize
+	if b%wordSize != 0 {
+		words += 1
+	}
```

Comment:
calculate `words` just once instead of on every NodeID since `limit` is a configured constant?

Reply:
limit is from confguration file